pkg/actors/actor.go

```diff
@@ -7,12 +7,60 @@ package actors
 
 import (
 	"sync"
+	"sync/atomic"
 	"time"
+
+	diag "github.com/dapr/dapr/pkg/diagnostics"
 )
 
 type actor struct {
-	lock         *sync.RWMutex
-	lastUsedTime time.Time
-	busy         bool
-	busyCh       chan (bool)
+	actorType string
+	actorID   string
+
+	concurrencyLock *sync.RWMutex
+	lastUsedTime    time.Time
+	busy            bool
+	busyCh          chan (bool)
+
+	pendingLockCount int32
+}
+
+func newActor(actorType, actorID string) *actor {
+	return &actor{
+		actorType:       actorType,
+		actorID:         actorID,
+		concurrencyLock: &sync.RWMutex{},
+		busy:            false,
```

Comment:
@yaron2 original code was `true` but, isn't `busy` field `false` when actor is added ? so after user request gets lock, it will change busy to true.

Reply:
Wit new code:
Lets say 2 calls for the same actor id come in parallel, an object would be created, both the calls will continue with the same object, lets say they enter lock() method at the same time, one of them will get the lock and mark the actor object busy busy.

With older code:
Now lets say 2 calls for the same actor id come in parallel, an object would be created and marked as busy, both the calls will continue with the same object, lets say they enter lock() method at the same time, one of them will get the lock.

So in both the scenarios once an actor call has the lock its busy, I don't see a difference in both, essentially it would be the same.